applications/credhub-api/src/main/resources/db/migration/mysql/V57_2__delete_related_encrypted_value_when_credential_is_deleted.sql

```diff
@@ -0,0 +1,50 @@
+-- Fix for https://github.com/cloudfoundry/credhub/issues/231
+-- Create delete trigger on credential table instead of the tables directly
+-- reference encrypted_value table because the trigger does not work for
+-- cascade-deleted records in MySQL.
+DROP TRIGGER IF EXISTS tr_credential_deleted;
+DELIMITER |
+CREATE TRIGGER tr_credential_deleted BEFORE DELETE ON credential
+FOR EACH ROW
+BEGIN
+  DROP TEMPORARY TABLE IF EXISTS
+    tmp_credential_version_uuids, tmp_encrypted_value_uuids;
```

Comment:
Are we sure that this can't delete the tables of another delete trigger running in parallel?

Reply:
Only if you have another trigger that uses the same tables, which we do not.

Reply:
I have confirmed that the MySQL temp table scope is per connection. So doing `credhub delete` from different client for example, won't be a problem. However, there is possibility that the process is triggered multiple times in the same connection as it is possible that multiple delete operations can be done in the same connection. I will re-work the trigger code to protect from that scenario.

Reply:
The trigger had been changed not to use temp tables.